lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/service/CriCheckingService.java

```diff
@@ -259,29 +260,40 @@ public JourneyResponse checkVcResponse(
         }
 
         if (configService.enabled(DL_AUTH_SOURCE_CHECK)
-                && getDcmawDlVc(newVcs).isPresent()
-                && getSuccessfulDlVc(sessionCredentials).isEmpty()) {
+                && requiresAuthoritativeSourceCheck(newVcs, sessionVcs)) {
             return JOURNEY_DL_AUTH_SOURCE_CHECK;
         }
 
         return JOURNEY_NEXT;
     }
 
-    private Optional<VerifiableCredential> getDcmawDlVc(List<VerifiableCredential> vcs) {
+    private boolean requiresAuthoritativeSourceCheck(
+            List<VerifiableCredential> newVcs, List<VerifiableCredential> sessionVcs) {
+        return extractDrivingPermitIdentifier(newVcs, Cri.DCMAW, false)
+                .map(
+                        identifier ->
+                                !identifier.equals(
+                                        extractDrivingPermitIdentifier(
+                                                        sessionVcs, Cri.DRIVING_LICENCE, true)
+                                                .orElse(null)))
+                .orElse(false);
+    }
+
+    private Optional<String> extractDrivingPermitIdentifier(
```

Comment:
This is a bit of a strange compound method - I wonder if it would be clearer to split into
- `Optional<VC> getVcFromCri(List<VC> vcs, Cri cri)`
- `String getDrivingPermitIdentifier(VC vc)`

(not massively important...)

I do like splitting out `requiresAuthoritativeSourceCheck` though :)